src/main/java/org/folio/dcb/service/impl/TransactionsServiceImpl.java

```diff
@@ -6,29 +6,56 @@
 import org.folio.dcb.domain.dto.TransactionStatus;
 import org.folio.dcb.domain.dto.TransactionStatusResponse;
 import org.folio.dcb.domain.entity.TransactionEntity;
+import org.folio.dcb.domain.mapper.TransactionMapper;
+import org.folio.dcb.exception.ResourceAlreadyExistException;
 import org.folio.dcb.repository.TransactionRepository;
 import org.folio.dcb.service.LibraryService;
 import org.folio.dcb.service.TransactionsService;
 import org.folio.spring.exception.NotFoundException;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Service;
 
+import java.util.Objects;
+
+import static org.folio.dcb.domain.dto.DcbTransaction.RoleEnum.BORROWER;
+import static org.folio.dcb.domain.dto.TransactionStatus.StatusEnum.CREATED;
+
 @Service
 @RequiredArgsConstructor
 @Log4j2
 public class TransactionsServiceImpl implements TransactionsService {
 
+  private static final String TEMP_VALUE_MATERIAL_TYPE_NAME_BOOK = "book";
+
   @Qualifier("lendingLibraryService")
   private final LibraryService lendingLibraryService;
+  @Qualifier("borrowingLibraryServiceImpl")
+  private final LibraryService borrowingLibraryServiceImpl;
   private final TransactionRepository transactionRepository;
+  private final TransactionMapper transactionMapper;
 
   @Override
   public TransactionStatusResponse createCirculationRequest(String dcbTransactionId, DcbTransaction dcbTransaction) {
     log.debug("createCirculationRequest:: creating new transaction request for role {} ", dcbTransaction.getRole());
-    return switch (dcbTransaction.getRole()) {
-      case LENDER -> lendingLibraryService.createTransaction(dcbTransactionId, dcbTransaction);
-      default -> throw new IllegalArgumentException("Other roles are not implemented");
-    };
+    checkTransactionExistsAndThrow(dcbTransactionId);
+
+    if (BORROWER == dcbTransaction.getRole()){
+      var virtualItem = dcbTransaction.getItem();
+      virtualItem.setPickupLocation("3a40852d-49fd-4df2-a1f9-6e2641a6e91f");   // leave it as a temporary solution. checked with Magzhan. Until the field-container will be added into DcbTransaction
+      virtualItem.setMaterialType(TEMP_VALUE_MATERIAL_TYPE_NAME_BOOK);
```

Comment:
As of now, there is a field in the DCBItem object as materialType. In this case, I think we don't need to have the above line.

Reply:
entire if-clause is currently a temporary solution.
I believe, I should leave the note about it.

Reply:
MaterialType should be taken from DCBItem, but currently, there is no chance of having it on the client's side.
So, this if-block is used as a temporary solution.

Reply:
changed